examples/cuda/integrand.cpp

```diff
@@ -0,0 +1,89 @@
+//#include "cuda_kernel.h"
+
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/framework/shape_inference.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "integrand.h"
+
+/* 
+ * In this example we follow the TF guide for operation creation
+ * https://www.tensorflow.org/guide/create_op
+ * to create an integrand as a custom operators.
+ *
+ * To first approximation, these operators are function that take
+ * a tensor and return a tensor.
+ */
+
+using namespace tensorflow;
+
+using GPUDevice = Eigen::GpuDevice;
+using CPUDevice = Eigen::ThreadPoolDevice;
+
+// CPU
+template <typename T>
+struct IntegrandOpFunctor<CPUDevice, T> {
+    void operator()(const CPUDevice &d, const T *input, T *output, const int nevents, const int dims) {
+        for (int i = 0; i < nevents; i++) {
+            output[i] = 0.0;
+            for(int j = 0; j < dims; j++) {
+                output[i] += input[i,j];
+            }
+        }
+    }
+};
+
+
+/* The input and output type must be coherent with the types used in tensorflow
+ * at this moment we are using float64 as default for vegasflow.
+ *
+ * The output shape is set to be (input_shape[0], ), i.e., number of events
+ */
+REGISTER_OP("IntegrandOp")
+    .Input("xarr: double")
+    .Output("ret: double")
+    .SetShapeFn(::tensorflow::shape_inference::UnchangedShape);
+        c -> set_output(0, c -> MakeShape( { c -> Dim(c -> input(0), 0) } ) );
```

Comment:
Ok, if you use my patch then you have to drop this and the next 2 lines.

Reply:
Oh, ok. I'll try it later and do the changes manually

Reply:
Nah, dropping this doesn't work either. I'll revert the patch because it doesn't really matter tbh ¯\_(ツ)_/¯

Reply:
It doesn't work because UnchangedShape supposes the output shape is identical to the input shape, and this is probably not the case everywhere? That sounds strange...

Reply:
Ah, no, no, it's not the case.
The input is (events, dimensions) and the output is only (dimensions,)

Reply:
Ok, then no way, your code is correct.